adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java

```diff
@@ -550,34 +550,36 @@ void onActivityResult(final int requestCode, final int resultCode, final Intent
                             "User cancelled the flow RequestId:" + requestId + correlationInfo));
                 } else if (resultCode == AuthenticationConstants.UIResponse.BROKER_REQUEST_RESUME) {
-                    Logger.v(TAG + methodName, "Device needs to have broker installed, waiting the broker "
+                    synchronized (this) {
+                        Logger.v(TAG + methodName, "Device needs to have broker installed, waiting the broker "
                             + "installation. Once broker is installed, request will be resumed and result "
                             + "will be received");
 
-                    //Register the broker resume result receiver with intent filter as broker_request_resume and
-                    // specific app package name
-                    mBrokerResumeResultReceiver = new BrokerResumeResultReceiver();
-                    (new ContextWrapper(mContext)).registerReceiver(mBrokerResumeResultReceiver,
-                            new IntentFilter(AuthenticationConstants.Broker.BROKER_REQUEST_RESUME
-                                    + mContext.getPackageName()), null, mHandler);
-
-                    // Send cancel result back to caller if doesn't receive result from broker within 5 minuites
-                    final int timoutForBrokerResult = 10 * 60 * 1000;
-                    mHandler.postDelayed(new Runnable() {
-
-                        @Override
-                        public void run() {
-                            if (!mBrokerResumeResultReceiver.isResultReceivedFromBroker()) {
-                                Logger.v(TAG + "onActivityResult", "BrokerResumeResultReceiver doesn't receive "
-                                        + "result from broker within 10 minuites, unregister the receiver and "
-                                        + "cancelling the request");
-
-                                (new ContextWrapper(mContext)).unregisterReceiver(mBrokerResumeResultReceiver);
-                                waitingRequestOnError(waitingRequest, requestId, new AuthenticationCancelError(
-                                        "Broker doesn't return back the result within 10 minuites"));
+                        //Register the broker resume result receiver with intent filter as broker_request_resume and
+                        // specific app package name
+                        mBrokerResumeResultReceiver = new BrokerResumeResultReceiver();
+                        (new ContextWrapper(mContext)).registerReceiver(mBrokerResumeResultReceiver,
+                                new IntentFilter(AuthenticationConstants.Broker.BROKER_REQUEST_RESUME
+                                        + mContext.getPackageName()), null, mHandler);
+    
+                        // Send cancel result back to caller if doesn't receive result from broker within 5 minuites
+                        final int timoutForBrokerResult = 10 * 60 * 1000;
```

Comment:
>10 * 60 * 1000 [](start = 58, length = 14)

comment says 5 minutes, this looks like 10 minutes, also can you move this to a private final static variable